src/java.base/share/classes/java/lang/foreign/FunctionDescriptor.java

```diff
@@ -107,8 +104,7 @@ public static FunctionDescriptor ofVoid(MemoryLayout... argLayouts) {
      * @return a variadic function descriptor, or this descriptor if {@code variadicLayouts.length == 0}.
      */
     public FunctionDescriptor asVariadic(MemoryLayout... variadicLayouts) {
-        Objects.requireNonNull(variadicLayouts);
-        Arrays.stream(variadicLayouts).forEach(Objects::requireNonNull);
+        assertVarargsNonNull(variadicLayouts);
         return variadicLayouts.length == 0 ? this : new VariadicFunction(this, variadicLayouts);
```

Comment:
We could change the VariadicFunction constructor to accept a List of non-null elements.

Reply:
It already throws an NPE if any of the args or variadicLayouts elements are `null`. So, the check is redundant also here.

Reply:
Well, it turned out it didn't. But soon it will...